### PR TITLE
Add generic smoke tests

### PR DESCRIPTION
**Pull request by chris-pateman**
### Change description ###
Add generic smoke tests so the pipeline will pass.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X ] No
```

### CODE REVIEW

File: src/smokeTest/resources/application.properties [hunk 1]
```diff
@@ -0,0 +1,6 @@
+spring.main.allow-bean-definition-overriding=true
```
Comment (timja) [inline src/smokeTest/resources/application.properties:1]: this should not be added in as a default, this means you've likely got a bug in your application.

File: src/smokeTest/java/uk/gov/hmcts/reform/pip/rules/smoketests/SmokeTestConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package uk.gov.hmcts.reform.pip.rules.smoketests;
+
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.PropertySource;
+
+@ComponentScan("uk.gov.hmcts.reform.pip.rules.smoketests")
+@PropertySource("application.properties")
+public class SmokeTestConfiguration {
```
Comment (timja) [inline src/smokeTest/java/uk/gov/hmcts/reform/pip/rules/smoketests/SmokeTestConfiguration.java:8]: is this enough to make the pipeline pass without and actual test?

this on it's own looks fine, although the tests are intentionally not added here as we don't want people to forget to add smoke tests. but I do believe the pipeline should pass first time.

Any thoughts?

Comment (chris-pateman) [reply src/smokeTest/java/uk/gov/hmcts/reform/pip/rules/smoketests/SmokeTestConfiguration.java:8]: it worked for the pip-business-rules, but didn't for hmi case hq, so might have spoken to soon.
I don't now Java, so this is a copy and paste job with a developers review.

If you don't think this is best then can close the PR

Comment (timja) [reply src/smokeTest/java/uk/gov/hmcts/reform/pip/rules/smoketests/SmokeTestConfiguration.java:8]: can you request their review on this PR, I'll also ping a couple of others.

Comment (adusumillipraveen) [reply src/smokeTest/java/uk/gov/hmcts/reform/pip/rules/smoketests/SmokeTestConfiguration.java:8]: Should we add a TODO to the Read me that team should add smoke tests based on functionality ? Just as a reminder for them.

Comment (chris-pateman) [reply src/smokeTest/java/uk/gov/hmcts/reform/pip/rules/smoketests/SmokeTestConfiguration.java:8]: @adusumillipraveen I think the pipeline needs the smoke test to run. It doesn't need actually tests within it, but it does need setting up.
There can get the generic set up of smoke test, but then will need a update for the folder structure and naming.

Comment (chris-pateman) [reply src/smokeTest/java/uk/gov/hmcts/reform/pip/rules/smoketests/SmokeTestConfiguration.java]: After correcting the folder paths, removing the flag that Tim mentioned and added some missing lines in the gradle build file it now works.
As well as this working in this project it also passes the build, which was failing in another project:
https://sds-build.platform.hmcts.net/blue/organizations/jenkins/HMCTS_HMI%2Fhmi-case-hq-emulator/detail/master/25/pipeline

the user would just need to update naming same as anything else

File: src/smokeTest/resources/application.properties [hunk 3]
```diff
@@ -0,0 +1,5 @@
+env=${ENVIRONMENT_NAME:local}
```
Comment (timja) [inline src/smokeTest/resources/application.properties:1]: shouldn't be needed
```suggestion
```

File: src/smokeTest/resources/application.properties [hunk 4]
```diff
@@ -0,0 +1,5 @@
...
+url=${TEST_URL:http://localhost:9000}
```
Comment (timja) [inline src/smokeTest/resources/application.properties:2]: this should be accessed as a spring property test.url, env override is automatic based on convention
```suggestion
test.url=http://localhost:8080
```

File: src/smokeTest/resources/application.properties [hunk 5]
```diff
@@ -0,0 +1,5 @@
...
+http.timeout=-1
+http.requestTimeout=-1
+http.readTimeout=-1
```
Comment (timja) [inline src/smokeTest/resources/application.properties:5]: why?
```suggestion
```

File: build.gradle [hunk 6]
```diff
@@ -203,7 +204,9 @@ dependencies {
 
   functionalTestImplementation sourceSets.main.runtimeClasspath
 
+  smokeTestImplementation group: 'io.rest-assured', name: 'rest-assured', version: versions.restAssured
```
Comment (timja) [inline build.gradle:207]: hmm, probably shouldn't add dependencies that are not used. you could add a ”real” test that hits the root endpoint with a TODO around it saying implement a real test?

**Review by timja — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 6 above)
Comment (chris-pateman) [reply build.gradle]: worked without this so removed.

**Review by timja — APPROVED** [review]
Thanks!

**Pull request merged**